src/hotspot/share/jfr/leakprofiler/checkpoint/objectSampleCheckpoint.cpp

```diff
@@ -197,6 +197,62 @@ static bool stack_trace_precondition(const ObjectSample* sample) {
   return sample->has_stack_trace_id() && !sample->is_dead();
 }
 
+class StackTraceChunkWriter {
```

Comment:
`StackTraceChunkWriter` contains essentially the  same logic as in `StackTraceBlobInstaller`. The difference is that it writes Stack traces to the given `JfrChunkWriter`.